lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1524,6 +1524,21 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
     spv::BuiltIn Builtin = spv::BuiltInPosition;
     if (!GV->hasName() || !getSPIRVBuiltin(GV->getName().str(), Builtin))
       return BVar;
+    if (static_cast<uint32_t>(Builtin) >= internal::BuiltInSubDeviceIDINTEL &&
+        static_cast<uint32_t>(Builtin) <=
+            internal::BuiltInMaxHWThreadIDPerSubDeviceINTEL) {
+      if (!BM->isAllowedToUseExtension(
+              ExtensionID::SPV_INTEL_hw_thread_queries)) {
+        std::stringstream SS;
+        SS << "Intel HW thread queries must be enabled by "
```

Comment:
Alternatively, in case if the extension is not enabled, SubDeviceIDINTEL and MaxHWThreadIDPerSubDeviceINTEL can be replaced with OpConstant carrying 0, and HWThreadIDINTEL can be replaced with either 0 or GlobalLineralId, WDYT?

Reply:
Or even HWThreadIDINTEL can also be replaced with 0.

Reply:
I think this is a bit of "magic". I would not feed SPIR-V with an extension to the device that doesn't support it at all. Or it should be documented somewhere with like... giant red letters.

Reply:
Agree